### ISSUE TITLE
Unable to use `jest.mock` with Wallaby

### ISSUE THREAD

**Issue by ariel7342** [wix/yoshi#860]
# 🐛 Bug Report

### 😯 Current Behavior

i'm getting failing tests when I use wallaby (there are no failing test in the ci or in the local terminal run)
Wallaby instrumentation is probably breaking the hoisting behaviour.
apparently the jest.mock is not hoisted in wallaby, therefore there are error which connect to the data

### 🤔 Expected Behavior

the wallaby and yoshi run suppose to console the same error, if there are no error in yoshi run there not suppose to be in the wallaby

### 🔦 Context 

when I develop I'm getting all the time errors that are not true, it distract me.  

### 💁 Possible Solution 

I looked up and found a solutions. the same issue is represents here : [link](https://github.com/wallabyjs/public/issues/872)
TL;DR
need to add to compilers :
```
...
compilers: {
    '**/*.js*': wallaby.compilers.babel({babelrc: true, presets: ['babel-preset-jest']})
},
```

### 💻 Code Sample

[git link to me repo](https://github.com/wix-private/ecom/tree/master/client/wixstores-client/wixstores-client-product-page)

### 🌍 Your Environment
```
System:
    OS: macOS 10.14.1
    CPU: (8) x64 Intel(R) Core(TM) i7-8559U CPU @ 2.70GHz
  Binaries:
    Node: 8.9.1 - ~/.nvm/versions/node/v8.9.1/bin/node
    Yarn: Not Found
    npm: 5.5.1 - ~/.nvm/versions/node/v8.9.1/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Browsers:
    Chrome: 69.0.3497.100
    Firefox: 62.0.2
    Safari: 12.0.1
  npmPackages:
    @storybook/core: Not Found
    @storybook/react: Not Found
    babel-preset-yoshi:  3.17.0 
    eslint-config-yoshi:  3.21.0 
    eslint-config-yoshi-base:  3.21.0 
    jest-environment-yoshi-bootstrap:  3.20.0 
    jest-environment-yoshi-puppeteer:  3.20.0 
    jest-yoshi-preset:  3.20.0 
    storybook: Not Found
    tslint-config-yoshi:  3.21.0 
    tslint-config-yoshi-base:  3.21.0 
    typescript: ^2.9.2 => 2.9.2 
    webpack:  4.26.1 
    yoshi-config:  3.20.0 
    yoshi-helpers:  3.20.0 
    yoshi-runtime:  1.0.41 

 Yoshi config
{
  "servers": {
    "cdn": {
      "port": 4200
    }
  },
  "externals": {
    "react": {
      "amd": "react",
      "umd": "react",
      "commonjs": "react",
      "commonjs2": "react",
      "root": "React"
    },
    "react-dom": {
      "amd": "reactDOM",
      "umd": "react-dom",
      "commonjs": "react-dom",
      "commonjs2": "react-dom",
      "root": "ReactDOM"
    },
    "lodash": {
      "amd": "lodash",
      "commonjs": "lodash",
      "commonjs2": "lodash",
      "root": "_"
    },
    "image-client-api": "imageClientSDK"
  },
  "entry": {
    "editorMode": "./editorMode.entry.tsx",
    "productPage": "./components/ProductPageApp/ProductPageApp.entry.js",
    "worker": "./viewerScript/viewerScript.entry.js",
    "mocks": "../test/mock/local-overrides/mocks.entry.ts"
  },
  "enhancedTpaStyle": true,
  "exports": "[name]",
  "umdNamedDefine": false,
  "hmr": true
}
```

**Comment by Schniz** [comment]
Woah! great catch @ariel7342! ⚡️ 

The solution you suggested should work (I hope), do you want to try that, and send a PR fixing this issue? seems like the right place for it is here:

https://github.com/wix/yoshi/blob/012242ab14cc7078d309452e370ccbc5cd6e3658/packages/yoshi/config/wallaby-jest.js#L5-L10

**Comment by Schniz** [comment]
@dmfilipenko, I haven't seen a PR regarding this issue - if you want to jump in and submit a quick fix - that would be very cool. 😸

**Comment by ariel7342** [comment]
The solution that I posted here didn't helped. maybe some thing with the version of babel

**Comment by dmfilipenko** [comment]
@ariel7342 same for me

**Comment by ariel7342** [comment]
what about it ??

**Comment by ArtemGovorov** [comment]
This should help: https://github.com/wallabyjs/public/issues/872#issuecomment-471917701

**Comment by meltedspark** [comment]
Yoshi guys, can you approve this is the way to go with `Yoshi 3.0.0`?

**Comment by audriuss** [comment]
Any news on this? Tried to add a preprocessors that @ArtemGovorov linked on comment, however no success. Here is our wallaby config using Yoshi common config: 

`const yoshiConfig = require('yoshi/config/wallaby-jest');

module.exports = function(wallaby) {
  const yoshiWallabyCommon = yoshiConfig(wallaby);

  return Object.assign({}, yoshiWallabyCommon, {
    // set to undefined to let Wallaby decide the number of processes based on the system's capacity
    workers: undefined,
    testFramework: 'jest',
    compilers: {
      '**/*.js?(x)': wallaby.compilers.babel({
        presets: ['@babel/react'],
        plugins: [
          '@babel/plugin-proposal-class-properties',
          '@babel/plugin-syntax-dynamic-import',
        ],
      }),
    },
  });
};`

**Comment by just-jeb** [comment]
@audriuss Are you on Yoshi 4? I have a working example [here](https://github.com/wix-private/site-transfer-success-dialog/blob/master/wallaby.js).

**Comment by ArtemGovorov** [comment]
If you are using Jest v24 (and higher), check out our new [Wallaby zero config feature](https://wallabyjs.com/docs/intro/config.html#automatic-configuration).

**Comment by audriuss** [comment]
@just-jeb , yeah, I've tried to do similar thing, with no success :/ Here's mine [config](https://github.com/wix-private/people-chrome-extension/pull/6/files#diff-4e87b2c6dcfa10beeaf6797eaa6c66ec). @ArtemGovorov wow, it works ok with zero/auto configuration. Thanks! However, it would be nice to find a way to work with yoshi wallaby config

**Comment by ranyitz** [comment]
Thanks @audriuss. It seems to me that this issue still exists, opening.

PRs welcome for solutions.
